v4/source/roles.yaml

```diff
@@ -1,3 +1,37 @@
+  "/roles":
+    get:
+      tags:
+        - roles
+      summary: Get a list of all the roles
+      description: |
+        ##### Permissions
+
+        `manage_system` permission is required.
+
+        __Minimum server version__: 4.9
```

Comment:
The minimum server version for this API would be the first version it appears in.  It looks like this will make it into 5.33